test/unit/pjax_fallback.js

```diff
@@ -313,6 +313,50 @@ asyncTest("POST with data object"+s, function() {
   })
 })
 
+asyncTest("GET with data array"+s, function() {
+  var frame = this.frame
+
+  this.loaded = function() {
+    equal(frame.location.pathname, "/env.html")
+    equal(frame.location.search, "?foo%5B%5D=bar&foo%5B%5D=baz")
+
+    var env = JSON.parse(frame.$("#env").text())
+    equal(env['REQUEST_METHOD'], "GET")
+    deepEqual(env['rack.request.query_hash']['foo'], ['bar', 'baz'])
```

Comment:
Sure, let's add the expectation that there's a `_pjax` param as well. Although it makes this test concerned about an unrelated feature (`_pjax` param being that feature), it's the easiest way to ensure that extra params aren't getting serialized that we never expected.